### PR TITLE
feat: add testnet support for funding pool 

### PR DESCRIPTION
**Pull request by NunoAlexandre**
This PR introduces the following changes:

- [x] Support switching between the `Local` and `Ropsten` Ethereum environments
  - [x] Offer toggle in the Settings
  - [x] Extend `WalletConnectSigner` so that it works both when used in both environments

As follow up issues found from running the feature in the Ropsten network:

- [x] Correct the token scaling, from and to the Radicle Contracts
- [x] Use a big-number lib that supports decimals. This was important to:
  - [x] Support setting a decimal value as weekly budget, top-up, and withdraw amounts
  - [x] Preserve precision when converting Radicle Tokens from and to the contracts.
- [x] Show a `Wrong Network` screen when the user switches their connected wallet's network
- [x] Improve the waiting experiences while transactions are to be included in a block
- [x] Documentation ++

### CODE REVIEW

File: ui/DesignSystem/Component/Wallet/WrongNetwork.svelte
Comment (NunoAlexandre) [inline ui/DesignSystem/Component/Wallet/WrongNetwork.svelte:45]: This screen needs to be designed/reviewed by @brandonhaslegs / @juliendonck

File: ui/Screen/Funding/Pool/Outgoing.svelte
Comment (NunoAlexandre) [inline ui/Screen/Funding/Pool/Outgoing.svelte:24]: Meaning, once the user approved the transaction onboarding their support plan, we need to wait for the transaction to be included in a block. Until then, we show the `GetStarted` screen but its button will now be disabled, showing that it is waiting for the tx.

File: ui/src/ethereum.ts
Comment (NunoAlexandre) [inline ui/src/ethereum.ts:18]:
```suggestion
// all `Environment`s. See `supportedNetwork` to learn which networks
```

File: ui/src/ethereum.ts
Comment (NunoAlexandre) [inline ui/src/ethereum.ts:5]:
```suggestion
// The Ethereum environment we support and can thus connect to.
```

File: ui/src/ethereum.ts
Comment (NunoAlexandre) [inline ui/src/ethereum.ts:55]: I am now hardcoding this since this value is well settled. However, it _can_ change in a new deployment of the contracts. Reading asynchronously from the contracts and have it widely accessible every time we want to convert a number sounds quite cumbersome so we have to discuss 1) whether it's worth the hassle and 2) if it is, what'd be the best way to implement it.

File: ui/src/ethereum.ts
Comment (NunoAlexandre) [inline ui/src/ethereum.ts:57]: Not really sure how to name this function, so I called it instinctively: converts a given big number to the scale that we, humans, (no offence meant) operate on.

File: ui/src/ethereum.ts
Comment (NunoAlexandre) [inline ui/src/ethereum.ts:63]:
```suggestion
  return ethers.BigNumber.from(big.mul(TOKEN_SCALE).round().toString());
```

File: ui/src/funding/contract.ts
Comment (NunoAlexandre) [inline ui/src/funding/contract.ts:25]: Since the contracts are immutable, and since pointing to a proxy address that resolves to the latest one might cause API breaks at this stage of the game, I'd figure we could hardcode them for now. We can discuss introducing proxy contracts as a follow-up issue.

File: ui/src/funding/contract.ts
Comment (NunoAlexandre) [inline ui/src/funding/contract.ts:38]: Note to self: might be good to use dependency-injection here instead.

**Review by NunoAlexandre — COMMENTED** [review]
Giving it a review myself and leaving hopefully helpful comments to the reviewers :surfing_man:

File: ui/src/wallet.ts
Comment (NunoAlexandre) [inline ui/src/wallet.ts:103]:
```suggestion
```

File: ui/src/wallet.ts
Comment (NunoAlexandre) [inline ui/src/wallet.ts:159]:
```suggestion
          balance,
```

File: ui/src/ethereum.ts
Comment (NunoAlexandre) [inline ui/src/ethereum.ts:19]:
```suggestion
// The ethereum networks we may parse from a connected wallet across
// all `Environment`s. See `supportedNetwork` to learn which networks
// each Environment supports.
```

File: ui/src/wallet.ts
Comment (NunoAlexandre) [inline ui/src/wallet.ts:62]: :information_source: This is the project id that Infura provides in the backend. We have a free account so it's fine to have it here. We can always create a new project when we need to be more cautious.

File: ui/src/wallet.ts
Comment (NunoAlexandre) [inline ui/src/wallet.ts:243]: So this is essentially configurable behaviour through arguments, which I am not usually a fan of. The alternative would be to have a dedicated implementation of `ethers.Signer` for the `Local` and another for the `Ropsten` environments, stripped away of configurable behaviour.

I didn't go that route since we only need the check once. Let me know if you would like us to revisit this :hand:

File: ui/src/funding/pool.ts
Comment (geigerzaehler) [inline ui/src/funding/pool.ts:247]: In the rest of the code we seem to use the term “weekly budget”.

File: ui/Modal/Funding/Pool/Withdraw.svelte
Comment (rudolfs) [inline ui/Modal/Funding/Pool/Withdraw.svelte:28]: Wouldn't it make more sense to do this?
```suggestion
const balance = $store?.data.unwrap()?.balance || Big(0);
```
And remove the `  let balance = Big(0);` above, since it's not used anywhere else?

File: ui/Screen/Funding/Pool/Outgoing/GetStarted.svelte
Comment (rudolfs) [inline ui/Screen/Funding/Pool/Outgoing/GetStarted.svelte:45]: We could wrap this in a `<Tooltip>` explaining the user why it is disabled. Our tooltips already support conditionality, so that when the `value` prop is not present it'll just render the button without a tooltip.

File: ui/src/ethereum.ts
Comment (rudolfs) [reply ui/src/ethereum.ts:57]: Perhaps `toHumanReadable`?

File: ui/src/transaction.ts
Comment (rudolfs) [inline ui/src/transaction.ts:384]: 👀

File: ui/src/wallet.ts
Comment (rudolfs) [inline ui/src/wallet.ts:72]:
```suggestion
  close: () => {
```

**Review by rudolfs — COMMENTED** [review]
Nice work!

Left some remarks inline.
Do we have to change this wording, now that we switched to a weekly distribution:

```
ui/Screen/Funding/Pool/Outgoing/Support.svelte
276:          To stop or pause your support, set the monthly contribution to 0 or
```

**Comment by rudolfs** [comment]
Bumped into this error while testing creating a new pool on the "local" network:

```
{
  "code": "UnkownTransactionFailure",
  "message": "Failed to onboard your pool: an unkown transaction error occurred",
  "stack": "invalid BigNumber string (argument=\"value\", value=\"1e+21\", code=INVALID_ARGUMENT, version=bignumber/5.0.13)"
}
```

Screenshot 2021-02-10 at 17 54 15

File: ui/src/ethereum.ts
Comment (NunoAlexandre) [reply ui/src/ethereum.ts:61]: Me too, actually. I just copied it from the Radicle Contracts. `fromHuman` works for me!

File: ui/src/funding/contract.ts
Comment (NunoAlexandre) [reply ui/src/funding/contract.ts:28]: I don't, it's just an overlook. This, before, was a hardcoded constant. Now it's a derived store and I forgot to rename it.

File: ui/src/funding/contract.ts
Comment (NunoAlexandre) [reply ui/src/funding/contract.ts:58]: Yes I agree. That's what this comment https://github.com/radicle-dev/radicle-upstream/pull/1582#discussion_r570380346 was about. Let's do it.

File: ui/src/wallet.ts
Comment (NunoAlexandre) [reply ui/src/wallet.ts:240]: It is indeed surprising but it's the way around as the code shows. That's how I got it to work for both environments, but I will double-check.

File: ui/src/wallet.ts
Comment (NunoAlexandre) [reply ui/src/wallet.ts:351]: Good one, will look into this.

File: ui/Screen/Funding/Pool/Outgoing/GetStarted.svelte
Comment (NunoAlexandre) [reply ui/Screen/Funding/Pool/Outgoing/GetStarted.svelte:45]: Wholeheartedly agree, as I have suggested that a few months back :b I will follow it up in a coming PR.

File: ui/src/ethereum.ts
Comment (NunoAlexandre) [reply ui/src/ethereum.ts:55]: Done!

File: ui/src/ethereum.ts
Comment (NunoAlexandre) [reply ui/src/ethereum.ts:57]: They are both 'readable', but not both at the 'human' scale. Will leave it as is if that's ok with you.

File: ui/src/wallet.ts
Comment (NunoAlexandre) [reply ui/src/wallet.ts:242]: I am not sure how to phrase it though in a way that adds more information than what the code already provides. Do you have a recommendation?

File: FUNDING.md
Comment (NunoAlexandre) [reply FUNDING.md:68]: Well, just deleted the whole section. I wrote this down before writing any code to make sense of having different app environments, etherereum environments, and ethereum networks.

If we see any one confused from being able to choose between `Ropsten` or `Local` we can always extend these docs.

File: ui/src/funding/pool.ts
Comment (NunoAlexandre) [reply ui/src/funding/pool.ts:247]: Thanks!

File: ui/Modal/Funding/Pool/Withdraw.svelte
Comment (NunoAlexandre) [reply ui/Modal/Funding/Pool/Withdraw.svelte:28]: I like that but then the editor will go :banana: saying that `balance` is not declared. Thanks anyway!

**Comment by NunoAlexandre** [comment]
> Bumped into this error while testing.
> It happened when I clicked the "Confirm in your wallet" button:

Thank you, @rudolfs. You found a critical issue, which is now fixed :hand:

File: ui/Screen/Funding/Pool/Outgoing/GetStarted.svelte
Comment (NunoAlexandre) [reply ui/Screen/Funding/Pool/Outgoing/GetStarted.svelte:45]: https://github.com/orgs/radicle-dev/projects/14#card-54705424

File: ui/src/wallet.ts
Comment (NunoAlexandre) [reply ui/src/wallet.ts:245]: :heavy_check_mark:

File: ui/src/funding/contract.ts
Comment (NunoAlexandre) [reply ui/src/funding/contract.ts]: Done :heavy_check_mark:

File: ui/src/wallet.ts
Comment (NunoAlexandre) [reply ui/src/wallet.ts]: Done.

**Comment by NunoAlexandre** [comment]
@rudolfs @geigerzaehler thank you for your throughout reviews. I have addressed or replied to everything, appreciate your second round :surfing_man:

File: ui/Modal/Funding/LinkAddress.svelte
Comment (geigerzaehler) [inline ui/Modal/Funding/LinkAddress.svelte:10]: Just a suggestion or something to consider for the future. If you use qualified imports like `import * as wallet` then you don’t need to alias exports and the import statement does not need to change if you are using more stuff.

File: ui/Screen/Funding/Pool/Outgoing.svelte
Comment (geigerzaehler) [inline ui/Screen/Funding/Pool/Outgoing.svelte:13]: We should avoid using `.subscribe` and use reactive statements (`$:`) instead. Explicit ubscriptions will need to be cleaned up or they leak memory and computation otherwise.

File: scripts/deploy-dev-contracts.js
Comment (NunoAlexandre) [reply scripts/deploy-dev-contracts.js:51]: Ahah dude, absolutely on the same side here. I initially called the other fn `toHumans` cause I didn't know what else to call it.

Maybe @cloudhead has a nice suggestion on how to name these functions?

File: ui/src/funding/contract.ts
Comment (NunoAlexandre) [reply ui/src/funding/contract.ts:36]: yes good point!

File: ui/Modal/Funding/LinkAddress.svelte
Comment (NunoAlexandre) [reply ui/Modal/Funding/LinkAddress.svelte:10]: Completly aware of that. The reason why I go this way for stores is that if you qualify the import, you can't do `$wallet.store` but you can do `$walletStore`.

File: ui/src/funding/contract.ts
Comment (NunoAlexandre) [reply ui/src/funding/contract.ts:36]: Actually, I prefer to keep it as is to avoid double dependencies on the site calls where these are used. I.e., if we drop this indirection, `wallet` needs to import `PoolContract` from `src/funding/contract` and `Erc20Factory` and `ERC20` from the Radicle Contracts.

File: ui/Screen/Funding/Pool/Outgoing.svelte
Comment (NunoAlexandre) [reply ui/Screen/Funding/Pool/Outgoing.svelte]: Done :)

**Comment by NunoAlexandre** [comment]
@geigerzaehler @rudolfs thanks again. Ready for the 3rd round :boxing_glove:

File: ui/src/transaction.ts
Comment (geigerzaehler) [inline ui/src/transaction.ts:225]: I think we can remove the `ongoing` function above now

Comment (NunoAlexandre) [reply ui/src/transaction.ts:225]: I am sorry, dropping it now.

**Comment by NunoAlexandre** [comment]
@rudolfs @geigerzaehler last round :pray:

**Review by geigerzaehler — APPROVED** [review]

**Comment by rudolfs** [comment]
Code looks good!

Found a couple of things while playing around with the local testnet. Don't know if they're due to the changes in this PR or not, so I won't block on this. Maybe worth investigating and doing some follow-ups after this is merged?

- After a while the funding tab gets really slow to load and there is no indication to the user that something is still loading. This happened after I manually advanced time via `await window.ethereumDebug.mineBlocks(10000);` a couple of times:

https://user-images.githubusercontent.com/158411/107758535-50b31d80-6d27-11eb-8672-2def4d674e3d.mov

- I can't seem to collect my incoming funds, I get an error about insufficient gas, although I have enough funds in my wallet and in the pool:
Screenshot 2021-02-12 at 11 33 15

```
{
  "code": "InsufficientGas",
  "message": "Failed to collect incoming support: you seem to have insufficient gas to cover this transaction.",
  "stack": "processing response error (body=\"{\\\"id\\\":296,\\\"jsonrpc\\\":\\\"2.0\\\",\\\"error\\\":{\\\"message\\\":\\\"Exceeds block gas limit\\\",\\\"code\\\":-32000,\\\"data\\\":{\\\"stack\\\":\\\"n: Exceeds block gas limit\\\\n    at S._queueTransaction (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:217389)\\\\n    at S.queueRawTransaction (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:215881)\\\\n    at p.eth_sendRawTransaction (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:205426)\\\\n    at p.handleRequest (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:202277)\\\\n    at t (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:70329)\\\\n    at a.handleRequest (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:200559)\\\\n    at t (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:70329)\\\\n    at f.s.handleRequest (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:183611)\\\\n    at f.handleRequest (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:181422)\\\\n    at t (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:70329)\\\\n    at o.handleRequest (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:189676)\\\\n    at t (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:70329)\\\\n    at a.handleRequest (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:189121)\\\\n    at t (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:70329)\\\\n    at u._handleAsync (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:70365)\\\\n    at Timeout._onTimeout (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:69793)\\\",\\\"name\\\":\\\"n\\\"}}}\", error={\"code\":-32000,\"data\":{\"stack\":\"n: Exceeds block gas limit\\n    at S._queueTransaction (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:217389)\\n    at S.queueRawTransaction (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:215881)\\n    at p.eth_sendRawTransaction (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:205426)\\n    at p.handleRequest (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:202277)\\n    at t (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:70329)\\n    at a.handleRequest (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:200559)\\n    at t (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:70329)\\n    at f.s.handleRequest (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:183611)\\n    at f.handleRequest (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:181422)\\n    at t (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:70329)\\n    at o.handleRequest (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:189676)\\n    at t (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:70329)\\n    at a.handleRequest (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:189121)\\n    at t (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:70329)\\n    at u._handleAsync (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:70365)\\n    at Timeout._onTimeout (/Users/rudolfs/work/radicle-upstream-2/node_modules/ganache-cli/build/ganache-core.node.cli.js:55:69793)\",\"name\":\"n\"}}, requestBody=\"{\\\"method\\\":\\\"eth_sendRawTransaction\\\",\\\"params\\\":[\\\"0xf869048504a817c800839fe40a9437c10d847bf9e708add451bf2f80c1297d7aa6918084e52253811ba05c715fc8fd27c8867ee9d9cc2e1fef80689f1dec3bce20316965e26cdae0e8aca078dadc2d029a9ae891ce483adac06e52c64ffff07c4fe17f16198fbdce8fd4c6\\\"],\\\"id\\\":296,\\\"jsonrpc\\\":\\\"2.0\\\"}\", requestMethod=\"POST\", url=\"http://localhost:8545\", code=SERVER_ERROR, version=web/5.0.12)"
}
```

**Review by rudolfs — APPROVED** [review]

**Comment by NunoAlexandre** [comment]
Merged in https://github.com/radicle-dev/radicle-upstream/commit/12b89267ad45d756733163cd0cf6175de351e5a6

**Pull request closed without merging**
